daemon/logger/journald/read.go

```diff
@@ -448,7 +450,7 @@ func (r *reader) signalReady() {
 	}
 }
 
-func (s *journald) ReadLogs(config logger.ReadConfig) *logger.LogWatcher {
+func (s *journald) ReadLogs(ctx context.Context, config logger.ReadConfig) *logger.LogWatcher {
```

Comment:
Plumbing contexts fully into the journald reader with cancelation is pretty straightforward: https://github.com/corhere/moby/commit/b603933dc10b8873455fe8c2d7489b23aa67d892
Feel free to cherry-pick that commit into this PR. Otherwise I would be happy to rebase and open a follow-up PR once this one has landed.

Reply:
Pulled in that commit (with some minor changes).